pkg/segment/results/mresults/seriesresult.go

```diff
@@ -286,6 +286,51 @@ func ApplyMathFunction(ts map[uint32]float64, function structs.Function) (map[ui
 		evaluate(ts, func(val float64) float64 {
 			return val * math.Pi / 180
 		})
+
+	case segutils.Acos:
+		err = evaluateWithErr(ts, func(val float64) (float64, error) {
+			if val < -1 || val > 1 {
+				return val, fmt.Errorf("evaluateWithErr: acos evaluate values in the range [-1,1]")
+			}
+			return math.Acos(val), nil
+		})
+	case segutils.Acosh:
+		err = evaluateWithErr(ts, func(val float64) (float64, error) {
+			if val < 1 {
+				return val, fmt.Errorf("evaluateWithErr: acosh evaluate values in the range [1,+Inf]")
+			}
+			return math.Acosh(val), nil
+		})
+	case segutils.Asin:
+		err = evaluateWithErr(ts, func(val float64) (float64, error) {
+			if val < -1 || val > 1 {
+				return val, fmt.Errorf("evaluateWithErr: asin evaluate values in the range [-1,1]")
+			}
+			return math.Asin(val), nil
+		})
+	case segutils.Asinh:
+		evaluate(ts, math.Asinh)
+	case segutils.Atan:
+		evaluate(ts, math.Atan)
+	case segutils.Atanh:
+		err = evaluateWithErr(ts, func(val float64) (float64, error) {
+			if val < -1 || val > 1 {
+				return val, fmt.Errorf("evaluateWithErr: atanh evaluate values in the range [-1,1]")
+			}
+			return math.Atanh(val), nil
+		})
```

Comment:
Got it! I just have a doubt. When I use a function to evaluate the value inside `timeSeries map[uint32]float64`, for some input, the function will return `NaN or Inf`. But since the map's value is `float64`, I cannot return `NaN or Inf` in this case. So I return an error instead. Is this the correct approach? What do you think?

Reply:
E.g.: Atanh(1) = +Inf, Atanh(-1) = -Inf
https://pkg.go.dev/math#Atanh

Reply:
I am curious because promql will return `NaN` in this case
<img width="798" alt="image" src="https://github.com/siglens/siglens/assets/67371917/918470eb-e2bd-4e0f-9ac8-76e6b8e669ba">

Reply:
Yes I think returning an error when the value is outside function's domain is a good approach. Otherwise the frontend has to somehow display NaNs to the user, which could be confusing.

Later if we get feedback from users that they'd rather get Nans and Infs rather than errors, we can change this behavior